ANMN/temperature_grid_product/anmn_temp_gridded_product.py

```diff
@@ -125,195 +61,174 @@ def get_var_var_qc_in_deployment(varname, nc_file_list):
+    depth_binned_array = []
+    
+    time_delta = (time_1d_interp[1]- time_1d_interp[0])
+        
+    time_bins_start = []
+    for j in range(n_time):
+        time_bins_start.append(time_1d_interp[j] - time_delta/2) # time_1d_interp sits in the centre of the bin
+
+    time_bins_start.append(time_1d_interp[j] + time_delta/2) # add last value
+
+    # histogram doesn't work with datetime so we need to use timestamps in seconds since a reference date
+    to_timestamp = np.vectorize(lambda x: (x - datetime(1970, 1, 1)).total_seconds())
+    
+    timestamp_bins_start = to_timestamp(time_bins_start)
+    
+    # temporal binning per dataset
+    for i_file in range(n_file):        
+        timestamp_values = to_timestamp(time_values[i_file])
+        
+        time_hist = np.histogram(timestamp_values, timestamp_bins_start)[0] # sometimes there is no data in a bin -> 0
+        
+        temp_binned  = np.histogram(timestamp_values, timestamp_bins_start, weights=temp_values[i_file]) [0] / time_hist # when there is no data in a bin -> 0 divided by 0 yields a NaN
+        depth_binned = np.histogram(timestamp_values, timestamp_bins_start, weights=depth_values[i_file])[0] / time_hist
+        
+        temp_binned_array.append(temp_binned)
+        depth_binned_array.append(depth_binned)
+
+    # vertical interpolation per time stamp
+    for j in range(n_time):
+        temp_binned  = np.array([row[j] for row in temp_binned_array])
+        depth_binned = np.array([row[j] for row in depth_binned_array])
+        
+        temp_binned  = temp_binned [~np.isnan(temp_binned)]
+        depth_binned = depth_binned[~np.isnan(depth_binned)]
+        
+        # we need to sort temp and depth by increasing depths before we can interpolate
+        temp_binned = [x for _,x in sorted(zip(depth_binned,temp_binned))]
+        depth_binned.sort()
+        
+        # we only want to interpolate what's between the depth_binned range, what is below or above is nan
+        temp_gridded[:,j] = np.interp(depth_1d_interp, depth_binned, temp_binned, left=np.nan, right=np.nan)
+    
+    return temp_gridded
+
+def list_instrument_meta(nc_file_list):
+    """ return a list of nominal_depth / sample_interval / serial_number gatt from the nc files"""
     instrument_nominal_depth = []
+    instrument_sample_interval = []
+    instrument_serial_number = []
     for f in nc_file_list:
         netcdf_file_obj = Dataset(f, 'r')
         instrument_nominal_depth.append(netcdf_file_obj.instrument_nominal_depth)
+        instrument_sample_interval.append(netcdf_file_obj.instrument_sample_interval)
+        instrument_serial_number.append(netcdf_file_obj.instrument_serial_number)
         netcdf_file_obj.close()
+    instrument_sample_interval = [x for _,x in sorted(zip(instrument_nominal_depth,instrument_sample_interval))]
+    instrument_serial_number = [x for _,x in sorted(zip(instrument_nominal_depth,instrument_serial_number))]
     instrument_nominal_depth.sort()
```

Comment:
Use `np.argsort(instrument_nominal_depth)` as in interpolation function above

Reply:
This only works on numpy arrays though, not on lists. Serial numbers are strings so not sure how you put that in a numpy array.